api/room.go

```diff
@@ -77,6 +77,33 @@ func (r *Room) Handle(c *websocket.Conn, l *LoginReq) error {
 			}
 			r.mu.Unlock()
 			r.broadcast(marshalChatBatch(cm.ToItem()))
+		case "kick_user":
+			msg, err := unmarshalAs[KickMessage](raw)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal kick message: %s", err)
+			}
+			if msg.User == l.User {
+				r.broadcast(marshalChatBatch(ChatBatchItem{
+					User:    "SYSTEM",
+					Text:    fmt.Sprintf("%s was trieed to kick himself", msg.User),
+					Created: time.Now().UTC().Format("2006-01-02 15:04:05"),
+				}))
+			} else {
+				if _, ok := r.Users[msg.User]; ok {
+					if err := r.Users[msg.User].C.WriteJSON(KickResp{Type: "kick_user", Reason: fmt.Sprintf("You were kicked by %s", l.User)}); err != nil {
+						return fmt.Errorf("failed to respond to kick_user: %s", err)
+					}
```

Comment:
1) the line is very long - really hard to read. let's extract some parts of it into tmp variables above (e.g. the KickResp message)
2) inside your `if` statement above you can declare a varibale for `r.Users[msg.User]` and use it here:

if u, ok := r.Users[msg.User]; ok {
    // `u` is now the user from the map
    ...
}
3) with this implementation we don't actually end the websocket connection with the user that got kicked out of the room. Consider closing the websocket after sending the kick message.